inference/models/gaze/gaze.py

```diff
@@ -0,0 +1,295 @@
+
+        return ret
+
+    def _make_response(self, faces: List[Detection], gazes: List[Tuple[float, float]], imgW: int, imgH: int,
+                       time_total: float, time_face_det: float = None, time_gaze_det: float = None) -> GazeDetectionInferenceResponse:
+        """Prepare response object from detected faces and corresponding gazes.
+
+        Args:
+            faces (List[Detection]): The detected faces.
+            gazes (List[tuple(float, float)]): The detected gazes (yaw, pitch).
+            imgW (int): The width (px) of original image.
+            imgH (int): The height (px) of original image.
+            time_total (float): The processing time.
+            time_face_det (float): The processing time.
+            time_gaze_det (float): The processing time.
+
+        Returns:
+            GazeDetectionInferenceResponse: The response object including the detected faces and gazes info.
+        """
+        predictions = []
+        for face, gaze in zip(faces, gazes):
+            landmarks = []
+            for keypoint in face.keypoints:
+                x = min(max(int(keypoint.x * imgW), 0), imgW - 1)
+                y = min(max(int(keypoint.y * imgH), 0), imgH - 1)
+                landmarks.append(Point(x=x, y=y))
+
+            bbox = face.bounding_box
+            x_center = bbox.origin_x + bbox.width / 2
+            y_center = bbox.origin_y + bbox.height / 2
+            score = face.categories[0].score
+
+            prediction = GazeDetectionPrediction(
+                face=FaceDetectionPrediction(x=x_center, y=y_center, width=bbox.width, height=bbox.height,
+                                             confidence=score, class_name="face", landmarks=landmarks),
+                yaw=gaze[0], pitch=gaze[1])
+            predictions.append(prediction)
+
+        response = GazeDetectionInferenceResponse(predictions=predictions, time=time_total, time_face_det=time_face_det, time_gaze_det=time_gaze_det)
+        return response
+
+    def get_infer_bucket_file_list(self) -> List[str]:
+        """Gets the list of files required for inference.
+
+        Returns:
+            List[str]: The list of file names.
+        """
+        return ["mediapipe_face_detector.tflite", "L2CSNet_gaze360_resnet50_90bins.onnx"]
+
+    def infer(self, request: GazeDetectionInferenceRequest) -> List[GazeDetectionInferenceResponse]:
+        """Detect faces and gazes in image(s).
+
+        Args:
+            request (GazeDetectionInferenceRequest): The request object containing the image.
+
+        Returns:
+            List[GazeDetectionInferenceResponse]: The list of response objects containing the faces and corresponding gazes.
+        """
+        if isinstance(request.image, list):
+            if len(request.image) > GAZE_MAX_BATCH_SIZE:
+                raise ValueError(
+                    f"The maximum number of images that can be embedded at once is {GAZE_MAX_BATCH_SIZE}"
```

Comment:
Update message to "The maximum number of images that can be inferred with gaze detection at one time is {GAZE_MAX_BATCH_SIZE}"